src/compiler/checker.ts

```diff
@@ -1160,6 +1160,13 @@ namespace ts {
 
                 if (!exportDefaultSymbol && !allowSyntheticDefaultImports) {
                     error(node.name, Diagnostics.Module_0_has_no_default_export, symbolToString(moduleSymbol));
+                    if (moduleSymbol.exports.has(node.symbol.name)) {
+                        diagnostics.add(createDiagnosticForNode(

```

Comment:
you should not add a new error. one of two options, either add an explanation to the main error using chainDiagnosticMessages, or have a new error message all together that is more specific to this case. 

I would go for the second one.